File: 03-sum-nat/sum_nat.erl
%% Erlang program to sum numbers from 1 to n
-module(sum_nat).
-author("M.V.Harish Kumar").
-compile([{date, "2023-12-07"}]).
-export([main/0]).

sum_to_n(0, Acc) -> Acc;
sum_to_n(N, Acc) -> sum_to_n(N-1, Acc+N).

sum_to_n(N) -> sum_to_n(N, 0).

main() ->
    {ok, [N]} = io:fread("Enter value for `n`: ", "~d"),
    Ans = sum_to_n(N),
    io:format("The sum from 1 to ~w is ~w~n", [N, Ans]),
    erlang:halt(0).

File: 03-sum-nat/sum_nat.exs
# Elixir program to sum numbers from 1 to n
# Created by M.V.Harish Kumar on 10-12-2023

defmodule Sum do
  def nat(0, acc) do
    acc
  end

  def nat(n, acc) do
    nat(n-1, acc+n)
  end

  def nat(n) do
    nat(n, 0)
  end
end

{n, _} = IO.gets("Enter value for `n`: ") |> Integer.parse
ans = Sum.nat(n)
IO.puts "The sum from 1 to #{n} is #{ans}"
